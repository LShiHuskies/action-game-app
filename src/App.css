.App {
  text-align: center;
  position: absolute;
  height: 100%;
  width: 100%;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  /* font-size: calc(10px + 2vmin); */
  /* font-size: 14px; */
  font-size: 1.5em;
  color: white;
}

.App-link {
  color: #61dafb;
}

.login {
  background-color: #C0C0C0;
  width: 500px;
  /* height: 330px; */
  margin-top: 0;
  border-radius: 2px;
  box-shadow: 0 12px 24px 0 rgba(0, 0, 0, 0.2);
}

#line{
  width: 440px;
  margin-top: 10px;
  border-color: purple;
  text-align: left;
  overflow: hidden;
  margin-left: 30px;
  margin-right: 30px;
  height: 12px;
  background-color: #e2e6eb;
  margin-bottom: 0px;
}

#demo {
  width: 150px;
  height: 50px;
  border-radius: 100px;
  text-align: center;
  float: left;
  color: #ffffff;
  margin-left: 30px;
  margin-top: 20px;
  font-size: 13px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.2px;
  margin-bottom: 20px;
  border: solid 1px #e2e6eb;
  padding-bottom: 1px;
  background-color: #282c34;
}

#submit {
  width: 150px;
  height: 50px;
  border-radius: 100px;
  text-align: center;
  float: right;
  color: #ffffff;
  margin-right: 30px;
  margin-top: 20px;
  font-size: 13px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.2px;
  margin-bottom: 20px;
  border: solid 1px #e2e6eb;
  padding-bottom: 1px;
  background-color: #282c34;
}

#inputText {
  height: 35px;
  border: solid 1px #e2e6eb;
  width: 150px;
  font-size: 14px;
  text-indent: 15px;
  font-weight: normal;
  font-style: normal;
  line-height: 1.43;
  letter-spacing: -0.2px;
  text-align: left;
  font-stretch: normal;
  margin-right: 25%;
}

#inputText input::placeholder{
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.43;
  letter-spacing: -0.2px;
  text-indent: 15px;
  opacity: 0.5;
  text-align: left;
}

.MuiFormControl-root.MuiTextField-root {
  width: 90%;
  margin-bottom: 10px;
}

.form_label {
  color: #282c34;
  float: left;
  margin-left: 20px;
  font-size: 16px;
  font-weight: bold;
  margin-bottom: 5px;
}

#SIGNUP {
  padding: 0;
  width: 100px;
  height: 34px;
  text-align: center;
  margin-right: 0;
  margin-top: 0;
  font-size: 11px;
  border-radius: 100px;
  border: solid 1px #e2e6eb;
  position: absolute;
  right: 2%;
  margin-left: 10px;
  background-color: #C0C0C0;
  display: inline;
  color: #282c34;
  border: solid, 1px, white;
  font-weight: bold;
  float: right;
}

#overview {
  font-size: 14px;
  padding: 7px;
  color:#C0C0C0;
  width: 300px;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: -0.2px;
  text-align: right;
  position: absolute;
  /* float: left; */
  top: 1.5%;
  display: inline;
  right: 1%;
}

#account_create {
  display: inline;
  position: absolute;
  top: 100%;
  left: 0%;
  right: 40%;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (max-width: 800px) and (max-height: 510px) {
  #overview {
    display: none !important;
  }
}


@media (max-width: 1050px) and (max-height: 510px) {
  #account_create{
    display: none !important;
  }
}