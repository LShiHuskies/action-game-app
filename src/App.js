import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import { connect } from 'react-redux';

import './App.css';
import Profile from './containers/profile';
import DynamicForm from './Reusable/dynamicForm';
import EmailForm from './containers/emailForm';
import MessageComp from './containers/message';
import SoloPlay from './containers/soloPlay';
import SinglePlayerGame from './containers/singlePlayerGame';

import Instructions from './components/instructions';



function App() {
  if (window.location.search.includes('token=')) {
    localStorage.setItem('token', window.location.search.split('token=')[1]);
  }
  return (
    <Router>
      <Switch>
        <Route exact path="/passwords/forgot" component={EmailForm}></Route>
        <Route exact path="/profile" component={Profile}></Route>
        <Route exact path="/signup" component={DynamicForm}></Route>
        <Route exact path="/activation-pending" component={MessageComp}></Route>
        <Route exact path="/solo" component={SoloPlay}></Route>
        <Route exact path="/instructions" component={Instructions}></Route>
        <Route exact path="/single-player" component={SinglePlayerGame}></Route>
        <Route path="/" component={DynamicForm}></Route>
      </Switch>
    </Router>
  );
}



export default connect(null)(App);
