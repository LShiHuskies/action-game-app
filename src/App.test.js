import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import App from './App';

import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';


const store = createStore(rootReducer, applyMiddleware(thunk));

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// test('render a login form on login', () => {
//   const { debug } = render(<Provider store={store}>
//     <App />
//   </Provider>);

//   console.log(debug());

//   const linkElement = screen.getByText(/submit/i);
//   expect(linkElement).toBeInTheDocument();
// });

// test('render a login form on login', () => {
//   const { debug } = render(<Provider store={store}>
//     <App />
//   </Provider>);

//   console.log(debug());

//   const linkElement = screen.getByText(/submit/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('when on default render of app', () => {

  it('should match snapshot', () => {
    const tree = renderer.create(<Provider store={store}>
      <App />
    </Provider>);
      expect(tree.toJSON()).toMatchSnapshot();
  });

});
