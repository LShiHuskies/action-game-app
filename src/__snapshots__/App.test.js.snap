// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when on default render of app should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="login"
    >
      
      <form
        onSubmit={[Function]}
      >
        <h1
          style={
            Object {
              "color": "#282c34",
              "marginTop": "0px",
              "paddingTop": "15px",
              "textAlign": "center",
            }
          }
        >
          Please Sign In
        </h1>
        <label
          className="form_label"
        >
          Username
        </label>
        <div
          className="MuiFormControl-root MuiTextField-root"
          input={
            Object {
              "name": "username",
              "onBlur": [Function],
              "onChange": [Function],
              "onDragStart": [Function],
              "onDrop": [Function],
              "onFocus": [Function],
              "value": "",
            }
          }
          meta={
            Object {
              "active": false,
              "asyncValidating": false,
              "autofilled": false,
              "dirty": false,
              "dispatch": [Function],
              "error": undefined,
              "form": "LoginForm",
              "initial": undefined,
              "invalid": false,
              "pristine": true,
              "submitFailed": false,
              "submitting": false,
              "touched": false,
              "valid": true,
              "visited": false,
              "warning": undefined,
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="standard-error-helper-text"
            id="standard-error-helper-text-label"
          >
            Username
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-describedby="standard-error-helper-text-helper-text"
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="standard-error-helper-text"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Username
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained"
            id="standard-error-helper-text-helper-text"
          >
            Please enter your username
          </p>
        </div>
        <label
          className="form_label"
        >
          Password
        </label>
        <div
          className="MuiFormControl-root MuiTextField-root"
          input={
            Object {
              "name": "password",
              "onBlur": [Function],
              "onChange": [Function],
              "onDragStart": [Function],
              "onDrop": [Function],
              "onFocus": [Function],
              "value": "",
            }
          }
          meta={
            Object {
              "active": false,
              "asyncValidating": false,
              "autofilled": false,
              "dirty": false,
              "dispatch": [Function],
              "error": undefined,
              "form": "LoginForm",
              "initial": undefined,
              "invalid": false,
              "pristine": true,
              "submitFailed": false,
              "submitting": false,
              "touched": false,
              "valid": true,
              "visited": false,
              "warning": undefined,
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="standard-error-helper-text"
            id="standard-error-helper-text-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-describedby="standard-error-helper-text-helper-text"
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="standard-error-helper-text"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained"
            id="standard-error-helper-text-helper-text"
          >
            Please enter your password
          </p>
        </div>
        <br />
        <p
          id="line"
          style={
            Object {
              "marginTop": "0px",
            }
          }
        >
          _______________________________________
        </p>
        <button
          id="demo"
          onClick={[Function]}
          type="button"
        >
          Forgot Username or Password
        </button>
        <button
          id="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      id="overview"
    >
      <div
        id="account_create"
      >
         
        Need to Create an Account?
         
      </div>
      <button
        id="SIGNUP"
        onClick={[Function]}
        type="submit"
      >
         
        Sign Up
         
      </button>
    </div>
  </header>
</div>
`;

exports[`when on default render of app should render the default login form and match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="login"
    >
      
      <form
        onSubmit={[Function]}
      >
        <h1
          style={
            Object {
              "color": "#282c34",
              "marginTop": "0px",
              "paddingTop": "15px",
              "textAlign": "center",
            }
          }
        >
          Please Sign In
        </h1>
        <label
          className="form_label"
        >
          Username
        </label>
        <div
          className="MuiFormControl-root MuiTextField-root"
          input={
            Object {
              "name": "username",
              "onBlur": [Function],
              "onChange": [Function],
              "onDragStart": [Function],
              "onDrop": [Function],
              "onFocus": [Function],
              "value": "",
            }
          }
          meta={
            Object {
              "active": false,
              "asyncValidating": false,
              "autofilled": false,
              "dirty": false,
              "dispatch": [Function],
              "error": undefined,
              "form": "LoginForm",
              "initial": undefined,
              "invalid": false,
              "pristine": true,
              "submitFailed": false,
              "submitting": false,
              "touched": false,
              "valid": true,
              "visited": false,
              "warning": undefined,
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="standard-error-helper-text"
            id="standard-error-helper-text-label"
          >
            Username
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-describedby="standard-error-helper-text-helper-text"
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="standard-error-helper-text"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Username
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained"
            id="standard-error-helper-text-helper-text"
          >
            Please enter your username
          </p>
        </div>
        <label
          className="form_label"
        >
          Password
        </label>
        <div
          className="MuiFormControl-root MuiTextField-root"
          input={
            Object {
              "name": "password",
              "onBlur": [Function],
              "onChange": [Function],
              "onDragStart": [Function],
              "onDrop": [Function],
              "onFocus": [Function],
              "value": "",
            }
          }
          meta={
            Object {
              "active": false,
              "asyncValidating": false,
              "autofilled": false,
              "dirty": false,
              "dispatch": [Function],
              "error": undefined,
              "form": "LoginForm",
              "initial": undefined,
              "invalid": false,
              "pristine": true,
              "submitFailed": false,
              "submitting": false,
              "touched": false,
              "valid": true,
              "visited": false,
              "warning": undefined,
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="standard-error-helper-text"
            id="standard-error-helper-text-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-describedby="standard-error-helper-text-helper-text"
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="standard-error-helper-text"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained"
            id="standard-error-helper-text-helper-text"
          >
            Please enter your password
          </p>
        </div>
        <br />
        <p
          id="line"
          style={
            Object {
              "marginTop": "0px",
            }
          }
        >
          _______________________________________
        </p>
        <button
          id="demo"
          onClick={[Function]}
          type="button"
        >
          Forgot Username or Password
        </button>
        <button
          id="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      id="overview"
    >
      <div
        id="account_create"
      >
         
        Need to Create an Account?
         
      </div>
      <button
        id="SIGNUP"
        onClick={[Function]}
        type="submit"
      >
         
        Sign Up
         
      </button>
    </div>
  </header>
</div>
`;
